tower-layer/src/util/never.rs

```diff
@@ -0,0 +1,14 @@
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug)]
+/// An error that can never occur.
+pub enum Never {}
```

Comment:
I've add this type here as it seems to be pretty useful for layer impls.

Reply:
I believe that there is a “void” crate for those that don’t want to implement a Never type.

Reply:
Not too sure if its worth it to bring in another crate for this?